src/napari_spatialdata/_view.py

```diff
@@ -115,12 +170,167 @@ def __init__(self, napari_viewer: Viewer, model: DataModel | None = None):
-            raise ValueError("Data points haven't been selected from the matplotlib visualisation.")
+        """Export selected points as adata column."""
+        if len(self.plot_widget.roi_list) > 0:
+
+            # display a message window to get the column name
+            self.annotation_name = self.open_annotation_dialog()
+
+            if self.annotation_name != "":
+
+                new_name = self.annotation_name not in self.model.adata.obs.columns
+
+                if not new_name:
+                    reply = QtWidgets.QMessageBox.question(
+                        self,
+                        "Overwrite Annotation",
+                        f"The annotation'{self.annotation_name}' already exists. Do you want to overwrite it?",
+                        QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No,
+                        QtWidgets.QMessageBox.No,
+                    )
+                    if reply == QtWidgets.QMessageBox.No:
+                        return
+
+                logger.debug(f"Annotating selected points as {self.annotation_name}")
+
+                # get selected points
+                self.selected_vector = self.plot_widget.get_selection()
+
+                # modify andata table
+                self.model.adata.obs[self.annotation_name] = self.selected_vector
+
+                # modify sdata of the viewer
+                if self._viewer is not None:
+                    selected_layer = self._viewer.layers.selection.active
+                    selected_table = self.table_name_widget.currentText()
+
+                    # change adata of the layer
+                    selected_layer.metadata["adata"] = self.model.adata
+
+                    # change sdata of the layer
+                    sel_obs = selected_layer.metadata["sdata"][selected_table].obs
+
+                    if not new_name:
+                        logger.debug("Dropping a column")
+                        # clear the old annotation column
+                        sel_obs = sel_obs.drop(self.annotation_name, axis=1)
+
+                    columns_to_add = [col for col in self.model.adata.obs.columns if col not in sel_obs.columns]
+                    merge_on = [
+                        self.model.adata.uns["spatialdata_attrs"]["region_key"],
+                        self.model.adata.uns["spatialdata_attrs"]["instance_key"],
+                    ]
+
+                    new_table = pd.merge(
+                        sel_obs, self.model.adata.obs[merge_on + columns_to_add], on=merge_on, how="left"
+                    )
+                    new_table[self.annotation_name] = new_table[self.annotation_name].astype("category")
+                    selected_layer.metadata["sdata"][selected_table].obs = new_table
+
+                    # trigger update of the viewer
+                    list(self._viewer.window._dock_widgets.items())[-2][1].children()[4]._on_layer_update()
```

Comment:
I would also replace the `[1]`, if possible, with something more explicit to avoid potential bugs like the one above. If this is not possible, I would add a comment explaining what `[1]` will access.

Reply:
Very good point and thank you for a detailed explanation! Fixed in 019c5189de0378ba62ddc82bc449e9dadf2caf5d